catenets/models/torch/snet.py

```diff
@@ -310,9 +310,22 @@ def _get_absolute_rowsums(mat: torch) -> torch.Tensor:
             return torch.sum(torch.abs(mat), dim=0)
 
         def _get_cos_reg(
-            params_0: torch.Tensor, params_1: torch.Tensor
+            params_0: torch.Tensor, params_1: torch.Tensor, normalize: bool = False
         ) -> torch.Tensor:
-            return torch.linalg.norm(torch.dot(params_0.T, params_1), "fro") ** 2
+            if normalize:
+                params_0 = params_0 / torch.linalg.norm(params_0, dim=0)
+                params_1 = params_1 / torch.linalg.norm(params_1, dim=0)
+
+            x_min = min(params_0.shape[0], params_1.shape[0])
+            y_min = min(params_0.shape[1], params_1.shape[1])
+            return (
+                torch.linalg.norm(
```

Comment:
I think this should stay  ``torch.linalg.norm(torch.dot(....), "fro")`` instead of the CosineSimilarity() functions. If not, the normalization doesn't make sense (cosine similarity normalizes internally, if I see this correctly). We can rename the function (get_cos_reg) to something else (e.g. get_fro_reg ; fro for frobenius) if you think its misleading because of the cosine in the name.